packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -195,27 +256,45 @@ const Cart = () => {
 
         return true
     }
+    /***************************** Update quantity **************************/
+
+    /***************************** Remove Item from basket **************************/
     const handleRemoveItem = async (product) => {
         setSelectedItem(product)
         setCartItemLoading(true)
-        try {
-            await basket.removeItemFromBasket(product.itemId)
-            toast({
-                title: formatMessage({
-                    defaultMessage: 'Item removed from cart',
-                    id: 'cart.info.removed_from_cart'
-                }),
-                status: 'success'
-            })
-        } catch {
-            showError()
-        } finally {
-            // reset the state
-            setCartItemLoading(false)
-            setSelectedItem(undefined)
-        }
+        await removeItemFromBasketAction.mutate(
+            {
+                parameters: {basketId: basket.basketId, itemId: product.itemId}
+            },
+            {
+                onSuccess: () => {
+                    // reset the state
+                    setCartItemLoading(false)
```

Comment:
With the removal of `finally` what is your base case condition where you `setCartItemLoading(false)` as seen in the original code?

```
} finally {
  // reset the state
  setCartItemLoading(false)
  setSelectedItem(undefined)
}
```

Reply:
good catch, I think we can use `onSettled` to reset the state here since this callback is a combination of both onSuccess and onError

Reply:
Did you say this was ready for review? Did you resolve this?

Reply:
Yes, I use onSettled to reset the state here. we still need onSuccess to show the toast, and error